### ISSUE TITLE
Pokemon Black freezes

### ISSUE THREAD

**Issue by briaguya-ai** [DS-Homebrew/nds-bootstrap#1746]
### Game Console(s) used

- [ ] Nintendo DS
- [ ] Nintendo DS Lite
- [ ] Nintendo DSi
- [X] Nintendo DSi XL/LL
- [ ] Nintendo 3DS
- [ ] Nintendo 3DS XL/LL
- [ ] New Nintendo 3DS XL/LL
- [ ] Nintendo 2DS
- [ ] New Nintendo 2DS XL/LL

### Launch Method

**TW**i**L**ight Menu++ SD Card

### Version used

nds-bootstrap Ver. Release v2.1.0
**TW**i**L**ight Menu++ v27.11.1

### Game tested/used/applicable

Pokemon Black (US/Europe) `md5: 37bff1431eda9b3a525737c7f59a432d`

### Expected Behavior

No freezes

### Actual Behavior

Freezing (input stops working, display keeps showing what it showed before, music continues)

### Steps to reproduce

This happened in a battle once when switching in a new Pokemon and once when moving between screens in the overworld.

### nds-bootstrap.ini

```ini
[NDS-BOOTSTRAP]
DONORTWLONLY0_NDS_PATH = nand:/title/00030005/484e4b4a/content/00000002.app
USE_ROM_REGION = 1
DEBUG = 0
LOGGING = 0
B4DS_MODE = 0
ROMREAD_LED = 0
DMA_ROMREAD_LED = -1
PRECISE_VOLUME_CONTROL = 0
SDNAND = 0
MACRO_MODE = 0
SLEEP_MODE = 1
SOUND_FREQ = 0
CONSOLE_MODEL = 0
HOTKEY = 284
NDS_PATH = sd:/Pokemon - Black Version (USA, Europe) (NDSi Enhanced).nds
SAV_PATH = sd:/saves/Pokemon - Black Version (USA, Europe) (NDSi Enhanced).sav
AP_FIX_PATH = sd:/_nds/nds-bootstrap/apFix.ips
RAM_DRIVE_PATH = sd:/null.img
GUI_LANGUAGE = en
LANGUAGE = 1
REGION = 1
DSI_MODE = 1
CARD_READ_DMA = 1
BOOST_CPU = 0
BOOST_VRAM = 0
ASYNC_CARD_READ = 0
DONOR_SDK_VER = 0
PATCH_MPU_REGION = 0
PATCH_MPU_SIZE = 0
FORCE_SLEEP_PATCH = 0
```

### Logs

I switched to playing directly off my actual cart after the second crash so I don't have any crashes in logs. I've since turned on logging and at some point I'll hop back in to try to repro the crash again. I will update this with logs if/when I can get a crash to happen.

### Other notes

C-Gear disabled the entire time

### Will you be available for future issues and/or replies?

Yes

**Comment by RocketRobz** [comment]
Sounds like there's some issue with either the SD card or SD slot.
Try reformatting the SD card: https://dsi.cfw.guide/sd-card-setup.html

**Comment by briaguya-ai** [comment]
Thanks for the quick response! I followed that guide when setting everything up about a week ago but I hadn't run F3 to test the card. I ran `f3write` and `f3read` without reformatting and found zero data loss.

That makes me worry it might be the reader in the DSi itself instead of the card.

I ordered a new SD card (non micro) so when that arrives I'll follow the guide to format and test write/read and move everything over to that.

Either way it sounds like this is not an issue with `nds-bootstrap` so I'll go ahead and close this.

Thanks again!
